samples/httpd/main.c

```diff
@@ -87,7 +87,7 @@ int main(void)
 
 #if USE_DHCP
 	debugPrint("Waiting for DHCP...\n");
-	while (g_pnetif->dhcp->state != DHCP_STATE_BOUND)
+	while (dhcp_supplied_address(g_pnetif) == 0)
```

Comment:
http://git.savannah.nongnu.org/cgit/lwip.git/commit/?id=1db9631e7d5dfd0a566f188d0f17fc017bad5af9 changed the API for this. Instead of having to touch the inner details of the DHCP implementation, there is a new function called `dhcp_supplied_address`, which is supposed to return 1 when an address was retrieved via DHCP. The commit I linked caused the breakage, the new API was introduced in a different one.

(http://git.savannah.nongnu.org/cgit/lwip.git/commit/?id=dc7340bbd5209d6fc2a6df5c23d9cadc0d4244c8 is also related to this)